inference-engine/tests/functional/plugin/myriad/subgraph_tests/topk_k_propagation.cpp

```diff
@@ -0,0 +1,107 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <common_test_utils/test_common.hpp>
+#include <ngraph/shape.hpp>
+#include <ngraph/type/element_type.hpp>
+#include <ngraph/function.hpp>
+#include <ngraph_functions/utils/ngraph_helpers.hpp>
+#include <ngraph/opsets/opset4.hpp>
+#include <vpu/ngraph/operations/dynamic_shape_resolver.hpp>
+#include <vpu/ngraph/operations/static_shape_topk.hpp>
+#include <vpu/ngraph/transformations/dynamic_to_static_shape_topk.hpp>
+#include <vpu/ngraph/transformations/dynamic_to_static_shape.hpp>
+
+namespace {
+
+class DynamicToStaticTopKPropagation : public CommonTestUtils::TestsCommon,
+        public testing::WithParamInterface<int64_t> {
+public:
+    void SetUp() override {
+        const auto& k = GetParam();
+
+        const auto data = std::make_shared<ngraph::opset4::Parameter>(ngraph::element::i64, ngraph::Shape{1000});
+        const auto realK = std::make_shared<ngraph::opset4::Parameter>(ngraph::element::i32, ngraph::Shape{1});
+        const auto maxK = ngraph::opset4::Constant::create(ngraph::element::i32, {1}, {k});
+
+        const auto concat = std::make_shared<ngraph::opset4::Concat>(ngraph::OutputVector{realK, maxK}, 0);
+
+        const auto reduceMin = std::make_shared<ngraph::opset4::ReduceMin>(concat, ngraph::opset4::Constant::create(ngraph::element::i32, {1}, {0}), false);
+        const auto builtSubgraph = buildSubgraph(reduceMin);
+
+        const auto dsr = std::make_shared<ngraph::vpu::op::DynamicShapeResolver>(data, realK);
+        const auto topK = std::make_shared<ngraph::opset4::TopK>(dsr, builtSubgraph, 0, "max", "value");
+
+        ngraph::ResultVector results{std::make_shared<ngraph::opset4::Result>(topK->output(0)),
+                                     std::make_shared<ngraph::opset4::Result>(topK->output(1))};
+        const auto function = std::make_shared<ngraph::Function>(results, ngraph::ParameterVector{data, realK}, "TopKPropagationOfK");
+        topK->set_output_type(0, dsr->get_input_element_type(0), ngraph::PartialShape::dynamic(1));
```

Comment:
I believe @Maxim-Doronin implemented the way to avoid such structures in testing code

Reply:
It's actually the same what is done in DSR_TestsCommon base class and in other tests on DTS.

Reply:
I mean there is a PR from @Maxim-Doronin that is intended to change the way tests are written. @Maxim-Doronin please confirm

Reply:
Yes, there is PR https://github.com/openvinotoolkit/openvino/pull/2118. But it hasn't been merged yet

Reply:
But Maxim's PR goes to master branch, while current one goes to release branch. I can apply your suggestion to the following PR to master branch (https://github.com/openvinotoolkit/openvino/pull/2184) and I think it will be a separate discussion there.

Reply:
I missed the point that you're going to release. Let's move with it this way